tests/ssl-opt.sh

```diff
@@ -1893,6 +1893,22 @@ run_test    "Authentication: client no cert, ssl3" \
             -C "! mbedtls_ssl_handshake returned" \
             -S "X509 - Certificate verification failed"
 
+# Tests for CA list suppression
+
+run_test    "Authentication: CA suppression disabled (default)" \
+            "$P_SRV debug_level=3 auth_mode=required" \
+            "$P_CLI crt_file=data_files/server6.crt \
+             key_file=data_files/server6.key" \
+            0 \
+            -s "requested DN"
+
+run_test    "Authentication: CA suppression enabled" \
+            "$P_SRV debug_level=3 auth_mode=required suppress_ca=1" \
+            "$P_CLI crt_file=data_files/server6.crt \
+             key_file=data_files/server6.key" \
+            0 \
+            -S "requested DN"
+
```

Comment:
These two test cases test the server behavior with a cooperating client, we should also test the client behavior (verify that it doesn't send a non-advertised certificate, and that it will send a certificate if the list from the server is empty), and the server behavor with a non-cooperating client (client that sends a certificate with a non-advertised CA).